doc/user_guide/io.rst

```diff
@@ -1030,6 +1032,162 @@ HDF5 file. All other keyword arguments will be passed to
 Note that the model and other secondary data artifacts linked to the signal are not
 written to the file but these can be implemented at a later stage.
 
+Nexus and general HDF
+---------------------
+
+Background
+^^^^^^^^^^
+'NeXus <https://www.nexusformat.org>'_ is a common data format orginally 
+developed by the neutron, x-ray communities. It is still being developed as 
+an international standard by scientists and programmers representing major 
+scientific facilities in order to facilitate greater cooperation in the analysis
+and visualization of data.
+Nexus uses a variety of classes to record data, values, 
+units and other experimental metadata associated with an experiment. 
+For specific types of experiments an Application Definition may exist which
+defines an agreed common layout that facilities can adhere to.
+ 
+Nexus metadata and data are stored in Hierarchical Data Format Files (HDF5) with
+a .nxs extension although standards HDF5 extensions are sometimes used.
+
+Files must use the ``.nxs``,``.hdf``,``.hdf5``,``.h5`` file extension 
+in order to use this io plugin.
+Using the ``.nxs`` extension will default to the Nexus loader 
+Using any other extension with Nexus reader will require the 
+format is specfied as Nexus in the loader 
+
+As the Nexus format uses HDF5 and needs to read data and metadata structured
+in different ways the loader is written to quite flexible and can also be used 
+to load any hdf5 based file.  
+
+Inspecting
+^^^^^^^^^^
+Looking in a Nexus file for specific metadatda or to see what data
+is stored in the file is often useful - .e.g to find
+what position a specific stage was at: 
+
+    >>> from hyperspy.io_plugins.nexus import get_metadata_in_file
+    >>> results=get_metadata_in_file("sample.hdf5",search_keys=["stage1_z"])
+    {'entry': {'instrument': {'scannables': {'stage1': {'stage1_z': {'value': -9.871700000000002,
+    'attrs': {'gda_field_name': 'stage1_z',
+    'local_name': 'stage1.stage1_z',
+    'target': '/entry/instrument/scannables/stage1/stage1_z',
+    'units': 'mm'}}}}}}} 
+    
+    >>> from hyperspy.io_plugins.nexus import get_data_in_file
+    >>> results=get_metadata_in_file("sample.hdf5",search_keys=["stage1_z"])
+    NXdata found
+    /entry/xsp3_addetector
+    /entry/xsp3_addetector_total
+    HDF datasets found
+    /entry/solstice_scan/keys/uniqueKeys
+    /entry/solstice_scan/scan_shape
+    Out[3]:
+    (['/entry/xsp3_addetector', '/entry/xsp3_addetector_total'],
+     ['/entry/solstice_scan/keys/uniqueKeys', '/entry/solstice_scan/scan_shape'])    
+    
+
+Reading
+^^^^^^^
+Nexus files can contain multiple datasets within the same file.
+HyperSpy supports reading in one or more datasets however Nexus
+can contain a vary large number of datasets.
```

Comment:
Left me asking so what? Is the point "...large number of datasets and so additional arguments allow the user to specify which data from the Nexus file is loaded."?

Reply:
Changed - problem is hyperspy returns a list of signals, I'd prefer a dict or dictionarybrowser .  We can have 20 or 30 signals or datasets and the order they're saved in might vary so if you want a specific signal you can't just use an index - you need to count which index it is or use a method to sort through the signal names.  Easier to only load the ones you want  - they are loaded in the order of the search keys